### PR TITLE
Removing HXP statics

### PR DESCRIPTION
**Pull request by MattTuttle**
I am removing HXP static variables. The changes are listed in MIGRATION.md. Here are the notable changes.

* Removed `HXP.scene` because Engine manages more than one scene.
  * This is probably the bulk of the changes since old HaxePunk expected a single scene.
  * Layers are now shown per scene in LayerList.
  * Engine has add/remove/replace instead of push/pop. This is in place of setting HXP.scene.
* Moved `HXP.renderingScene` to `Engine.activeScene` and allowed it to be set during update as well as render.
* Detangled the resize code. Window resizing is handled by Engine and virtual screen resizing is handled by Screen.
  * Also, setting `Screen.scaleMode` will flag for a resize.
* Added a Timer class, moved `HXP.timeFlag` and `flash.Lib.getTimer`.
* Changed `Touch.sceneX/Y` to `Touch.cameraOffset`.
* Removed `HXP.halfWidth` and `HXP.halfHeight`. These are easily calculated by the user.

### CODE REVIEW

**Comment by bendmorris** [comment]
Going to go through this and test later today - great change 👍

**Comment by MattTuttle** [comment]
Let me know if you think of a better way to handle adding/removing scenes from Engine. That part still feels a bit flaky.

**Comment by ibilon** [comment]
Saw you removed `halfWidth` and `halfHeight`, don't know if it'd be worth it but you could change the type of `width`/`height` to
```haxe
abstract FloatPlus(Float) from Float to Float {
  public var half(get, never):Float;
  inline function get_half() return this/2;
}
```
So you could do `HXP.width.half`, and it has no runtime cost.

It's mostly for convenience and could be applied to other places, but like you said it's easily calculated by the user.

**Comment by bendmorris** [comment]
That's an idea, but `width/2` is shorter and less surprising than `width.half`. I don't think it's all that useful.

**Comment by bendmorris** [comment]
I'm not sure about the adding/removing/replacing scenes - what would usage normally look like? The current system, with a stack + a pointer to the topmost scene, has worked well for me. Maybe keeping push/pop/scene but adding replace would be better?

**Comment by MattTuttle** [comment]
I figured out a way to handle popScene. It's a bit more costly but should work the same. The trick is not updating the `_scenes` variable because it could be used in an iterator.

I also added several unit tests to make sure the functionality works as expected.

File: haxepunk/utils/DrawContext.hx [hunk 1]
```diff
@@ -384,11 +400,17 @@ class DrawContext
...
 	@:access(haxepunk.graphics.hardware.SceneRenderer)
-	inline function begin()
+	@:access(haxepunk.graphics.hardware.DrawCommand)
+	function begin()
 	{
-		if (shader == null) shader = new ColorShader();
-		var scene = (this.scene == null) ? (HXP.renderingScene == null ? HXP.scene : HXP.renderingScene) : this.scene;
-		command = scene.renderer.batch.getDrawCommand(null, shader, smooth, blend, null);
+		var scene = this.scene.or(HXP.engine.activeScene.ensure());
+		if (needsCommand || scene != lastScene || command._next != null)
```
Comment (MattTuttle) [inline haxepunk/utils/DrawContext.hx]: Something doesn't work with this check. It reduces the number of `getDrawCommand` calls but occasionally causes Draw commands to not render. I'm going to have to look into this before merging this code.

**Comment by MattTuttle** [comment]
Closing because it is old and I plan to do this over time

**Pull request closed without merging**